file-formats/clas/type/zif_oo_aff_clas_v1.intf.abap

```diff
@@ -0,0 +1,172 @@
+      "! Behavior class (behavior implementations)
+      behavior_class                 TYPE ty_category VALUE '06',
+    END OF co_category.
+
+  TYPES:
+    "! <p class="shorttext">Name and Description</p>
+    "! Name and description
+    BEGIN OF ty_component_description,
+      "! <p class="shorttext">Name</p>
+      "! Name
+      "! $required
+      name        TYPE zif_aff_types_v1=>ty_object_name_30,
+      "! <p class="shorttext">Description</p>
+      "! Description
+      "! $required
+      description TYPE zif_aff_types_v1=>ty_description_60,
+    END OF ty_component_description,
+
+    "! <p class="shorttext">Component Descriptions</p>
+    "! Class component descriptions
+    ty_component_descriptions TYPE SORTED TABLE OF ty_component_description WITH UNIQUE KEY name.
+
+
+  TYPES:
+    "! <p class="shorttext">Method Description</p>
+    "! Method description
+    BEGIN OF ty_method,
+      "! <p class="shorttext">Method Name</p>
+      "! Name of the method
+      "! $required
+      name        TYPE zif_aff_types_v1=>ty_object_name_30,
+      "! <p class="shorttext">Method Description</p>
+      "! Description of the method
+      "! $required
+      description TYPE zif_aff_types_v1=>ty_description_60,
+      "! <p class="shorttext">Parameter Descriptions</p>
+      "! Parameter descriptions
+      parameters  TYPE ty_component_descriptions,
+      "! <p class="shorttext">Exception Descriptions</p>
+      "! Exception descriptions
+      exceptions  TYPE ty_component_descriptions,
+    END OF ty_method,
+    "! <p class="shorttext">Method Descriptions</p>
+    "! Method descriptions
+    ty_methods TYPE SORTED TABLE OF ty_method WITH UNIQUE KEY name.
+
+  TYPES:
+    "! <p class="shorttext">Event Description</p>
+    "! Event description
+    BEGIN OF ty_event,
+      "! <p class="shorttext">Event Name</p>
+      "! Name of the event
+      "! $required
+      name        TYPE zif_aff_types_v1=>ty_object_name_30,
+      "! <p class="shorttext">Event Description</p>
+      "! Description of the event
+      "! $required
+      description TYPE zif_aff_types_v1=>ty_description_60,
+      "! <p class="shorttext">Parameter Descriptions</p>
+      "! Parameter descriptions
+      parameters  TYPE ty_component_descriptions,
+    END OF ty_event,
+    "! <p class="shorttext">Event Descriptions</p>
+    "! Event descriptions
+    ty_events TYPE SORTED TABLE OF ty_event WITH UNIQUE KEY name.
+
+  TYPES:
+    "! <p class="shorttext">Interface Properties</p>
+    "! Interface properties
+    BEGIN OF ty_main,
+      "! <p class="shorttext">Schema</p>
+      "! Format version
+      "! $required
+      schema               TYPE string,
+      "! <p class="shorttext">Header</p>
+      "! header
+      "! $required
+      header               TYPE zif_aff_types_v1=>ty_header_60_src,
+      "! <p class="shorttext">Class Category</p>
+      "! Class category
+      "! $required
```

Comment:
In interfaces the category was not required, maybe we should keep this in sync.

Reply:
I deleted the $required statement